raiden/api/python.py

```diff
@@ -645,6 +652,30 @@ def transfer_async(
         if not is_binary_address(target):
             raise InvalidAddress('target address is not valid.')
 
+        if secret is not None:
+            if len(secret) != 66:
+                raise InvalidSecretOrSecretHash('secret length should be 66.')
+            if not is_hex(secret):
+                raise InvalidSecretOrSecretHash('provided secret is not an hexadecimal string.')
+            secret = to_bytes(hexstr=secret)
+
+        if secret_hash is not None:
+            if len(secret_hash) != 66:
+                raise InvalidSecretOrSecretHash('secret_hash length should be 64.')
+            if not is_hex(secret_hash):
+                raise InvalidSecretOrSecretHash('secret_hash is not an hexadecimal string.')
+            secret_hash = to_bytes(hexstr=secret_hash)
+
+        if secret is None and secret_hash is None:
+            secret = random_secret()
+            secret_hash = sha3(secret)
+        elif secret is not None and secret_hash is None:
+            secret_hash = sha3(secret)
```

Comment:
This is redundant and will be done in raiden_service line 779 in this PR.

Reply:
Agree